src/radio-settings.c

```diff
@@ -391,41 +397,30 @@ static void radio_send_properties_reply(struct ofono_radio_settings *rs)
 	__ofono_dbus_pending_reply(&rs->pending, reply);
 }
 
-static void radio_set_modem_rats(struct ofono_radio_settings *rs,
-					const ofono_bool_t rats[])
-{
-	memcpy(rs->modem_rats, rats, sizeof(rs->modem_rats));
-	rs->modem_rats_filled = TRUE;
-}
-
-static void radio_modem_rats_query_callback(const struct ofono_error *error,
-						const ofono_bool_t rats[],
+static void radio_available_rats_query_callback(const struct ofono_error *error,
+						unsigned int available_rats,
 						void *data)
 {
 	struct ofono_radio_settings *rs = data;
-	DBusMessage *reply;
-
-	if (error->type != OFONO_ERROR_TYPE_NO_ERROR) {
-		DBG("Error during modem rats query");
 
-		reply = __ofono_error_failed(rs->pending);
-		__ofono_dbus_pending_reply(&rs->pending, reply);
-
-		return;
-	}
+	if (error->type == OFONO_ERROR_TYPE_NO_ERROR)
+		rs->available_rats = available_rats & 0x7;
+	else
+		DBG("Error while querying available rats");
 
-	radio_set_modem_rats(rs, rats);
 	radio_send_properties_reply(rs);
 }
 
-static void radio_query_modem_rats(struct ofono_radio_settings *rs)
+static void radio_query_available_rats(struct ofono_radio_settings *rs)
 {
-	if (rs->driver->query_modem_rats == NULL) {
+	/* Modem technology is not supposed to change, so one query is enough */
```

Comment:
I think this is OK, but just wanted to check...  When we change which SIM is capable of 3G on krillin, the modems are torn down and re-created, so this assumption ( ie. that modem technology doesn't change ) is still true, correct?